### PR TITLE
use accelerate autocast in jit eval path, since mix precision logic is…

### PR DESCRIPTION
**Pull request by sywangyi**
… in accelerator currently

Fixes # (issue)
mix precision logic is all moved to accelerator, so autocast_smart_context_manager does not take effect any longer. use accelerate autocast instead

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

- trainer: @sgugger

### CODE REVIEW

**Comment by sywangyi** [comment]
@yao-matrix

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -1265,9 +1265,14 @@ def torch_jit_model_eval(self, model, dataloader, training=False):
             example_batch = next(iter(dataloader))
             example_batch = self._prepare_inputs(example_batch)
             try:
-                jit_model = model.eval()
-                with ContextManagers([self.autocast_smart_context_manager(cache_enabled=False), torch.no_grad()]):
-                    if version.parse(version.parse(torch.__version__).base_version) >= version.parse("1.14.0"):
+                jit_model = copy.copy(model)
+                jit_model.eval()
+                original_forward = jit_model.__dict__.pop("_original_forward", None)
```
Comment (sgugger) [inline src/transformers/trainer.py:1270]: There is a method in accelerate to remove that hook (using `unwrap_model`). Might be better to use that here.

**Review by sgugger — COMMENTED** [review]
Thanks for your PR, suggested a couple of comments.

File: src/transformers/trainer.py (hunk 1 above)
Comment (sywangyi) [reply src/transformers/trainer.py:1270]: Hi, @sgugger . Thanks for your suggestion, I also try to use unwarp_model like jit_model = self.accelerator.unwrap_model(jit_model, keep_fp32_wrapper=False) in the code. but the forward function is not overridden as my expected. Take BERT QA as example, I expect the forward should be "bound method BertForQuestionAnswering.forward of BertForQuestionAnswering", but the actual forward is "function BertForQuestionAnswering.forward at 0x7f93a22235e0" , which will cause jit failure like "failed to use PyTorch jit mode due to: forward() missing 1 required positional argument: 'self'."

Comment (sywangyi) [reply src/transformers/trainer.py:1270]: I think it's caused by https://github.com/huggingface/accelerate/blob/main/src/accelerate/utils/other.py#L74-L78

Comment (sgugger) [reply src/transformers/trainer.py:1270]: Oh ok then.

**Review by sgugger — APPROVED** [review]

**Pull request merged**